.github/workflows/render_readme.yaml

```diff
@@ -0,0 +1,51 @@
+# Name of the workflow
+name: render-readme
+
+# Controls when the action will run. Triggers include:
+#
+# - button trigger from github action page
+# - on changes to readme.Rmd
+
+on:
+  workflow_dispatch:
+  push:
+    paths:
+      - 'README.Rmd'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  render-readme:
+    runs-on: macos-latest
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - name: Checkout repos
+        uses: actions/checkout@v3
+
+      - name: Setup R
+        uses: r-lib/actions/setup-r@v2
+
+      - name: Setup pandoc
+        uses: r-lib/actions/setup-pandoc@v2
+
+      - name: Install dependencies
+        uses: r-lib/actions/setup-r-dependencies@v2
+        with:
+          extra-packages: any::rmarkdown, local::.
+
+      - name: Compile the readme
+        run: |
+          rmarkdown::render("README.Rmd")
+        shell: Rscript {0}
+
+      - name: Commit files
+        run: |
+          git config --local user.email "action@github.com"
+          git config --local user.name "GitHub Action"
+          git add README.md 
+          git diff-index --quiet HEAD || git commit -m "Automatic readme update"
+          git push origin || echo "No changes to push"
```

Comment:
see https://github.com/epiforecasts/stackr/pull/19

I think as is this will cause failing actions because of push restrictions to `main`

Reply:
So @sbfnk you're suggestion would be to change the following? 
```suggestion
          status -sb | grep -q ahead && git push origin
```
But that wouldn't address push restrictions, right? 

(Admittedly I don't really know how they work anyway. I could - which of course I never would do :eyes: - push to main without a problem)

Reply:
yes - if the README is updated in a PR the action would then not fail on `main` as there would be nothing left to update.

Reply:
> I could in the past

You might have an exception for code owners or something like that

Reply:
What a dangerous exception - I'm the prime danger this branch needs to be protected from!

Reply:
Its because there are very few branch protections on this repo and even those can be ignored by admins which is all of us. It will likely tell you when you push to main that you are bypassing branch protections. Both those things can be changed but in the past you have given off strong I want to push to main vibes hence the current settings.

Reply:
More like strong "oooops I accidentally pushed to main vibes" 😆